fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -681,7 +682,43 @@ private ParameterDAO makeParameterDAO(Connection connection) throws FHIRPersiste
                         }
                     } else {
                         unsortedResultsList = resourceDao.search(query);
-                        resources = this.convertResourceDTOList(unsortedResultsList, resourceType, elements);
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> matchResultList = unsortedResultsList;
+                        List<com.ibm.fhir.persistence.jdbc.dto.Resource> includeResultList = new ArrayList<>();
+
+                        // Check if _include or _revinclude search. If so, remove duplicates from 'include' resources
+                        // (duplicates of both 'match' and 'include' resources) and make sure _elements processing
+                        // is not done for 'include' resources.
+                        if (searchContext.hasIncludeParameters() || searchContext.hasRevIncludeParameters()) {
+                            // Calculate 'match' results count
+                            int pageSize = searchContext.getPageSize();
+                            int offset = (searchContext.getPageNumber() - 1) * pageSize;
+                            int matchResultCount = pageSize;
+                            if (searchResultCount < offset + pageSize) {
+                                matchResultCount = searchResultCount - offset;
+                            }
+
+                            // Split results list into 'match' list and 'include' list
+                            if (unsortedResultsList.size() > matchResultCount) {
+                                matchResultList = unsortedResultsList.subList(0, matchResultCount);
+
+                                // Remove duplicates from 'include' list
+                                Set<Long> resultIds = new HashSet<>();
+                                for (com.ibm.fhir.persistence.jdbc.dto.Resource resource : matchResultList) {
+                                    resultIds.add(resource.getId());
+                                }
+                                for (int i=matchResultCount; i<unsortedResultsList.size(); ++i) {
+                                    com.ibm.fhir.persistence.jdbc.dto.Resource resource = unsortedResultsList.get(i);
+                                    if (!resultIds.contains(resource.getId())) {
```

Comment:
Question:  if a resource is both a `match` *and* and `include`, should we return it just once (as a match), or twice (once as a match and once as an include)?
This code seems to support the former:  a given resource will only be in the results once and if its both a match and an include then it will only be labeled as a match.

Reply:
Yes, I went by the tenet that a resource should only appear once in a search result bundle. Can certainly go the other way if that's the consensus.

Reply:
I cannot find guidance in the spec of which option is preferred.  Give that, I think the approach that a resource should only appear once in a search result bundle is a good one.

Reply:
I'm good with it too, just thought it was an important decision worth pondering (and maybe including somewhere either in the design doc or in our conformance.md doc)

Reply:
I documented in the conformance doc.